File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        stringstream ss(sentence);
        string word;
        int n = searchWord.size();
        int i = 1;
        while(ss >> word) {
            if(word.size() >= n && word.substr(0,n) == searchWord) return i;
            i++;
        }
        return -1;
    }
};

File: 1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution(object):
    def isPrefixOfWord(self, sentence, searchWord):
        string = sentence.split()
        n = len(searchWord)
        ans = -1
        for i in range(len(string)):
            if(len(string[i]) >= n):
                if(string[i][:n] == searchWord):
                    ans = i+1
                    break
        return ans

        